server.py

```diff
@@ -159,6 +159,27 @@ def format_raw_course(raw_course):
 def schedule_sort_key(slot):
     return slot["days"], slot["startTime"], slot["endTime"]
 
+def time_slot_equal(slot_a, slot_b):
+    return slot_a["days"] == slot_b["days"] and \
+           slot_a["location"] == slot_b["location"] and \
+           slot_a["startTime"] == slot_b["startTime"] and \
+           slot_a["endTime"] == slot_b["endTime"]
+
+def schedule_unique(sorted_schedule):
+    '''
+    Return a new schedule of unique timeslots in sorted_schedule
+    (sorted order retained).
+    '''
+    if sorted_schedule == [] :
+        return []
+    new_schedule = [sorted_schedule[0]]
+    for i in range(1, len(sorted_schedule)):
+        time_slot_last = sorted_schedule[i-1]
+        time_slot_i = sorted_schedule[i]
+        if not time_slot_equal(time_slot_last, time_slot_i) :
+            new_schedule += [time_slot_i]
+    return new_schedule
+
```

Comment:
As a matter of fact, why is this function necessary either? Why can't we just use `sorted(set(schedule), key=schedule_sort_key)`?

Reply:
Python dict is not mutable so it cannot be used as key (cannot be used in set).

Reply:
Python dict is not mutable so it cannot be used as key (cannot be used in set).

Reply:
I simplify the unique method to make it easier to read. The size of the schedule is always small. I don't know why I tried to write an O(n) solution back then.